javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -684,27 +685,55 @@ private void addModelConstructor(ClientModel model, JavaVisibility constructorVi
         // Also get required properties from the super class structure.
         List<ClientModelProperty> requiredParentProperties = ClientModelUtil.getRequiredWritableParentProperties(model);
 
+        // Jackson requires a constructor with @JsonCreator, with parameters in wire type. Ref https://github.com/Azure/autorest.java/issues/2170
+        boolean generatePrivateConstructorForJackson = false;
+
         // Description for the class is always the same, not matter whether there are required properties.
         // If there are required properties, the required properties will extend the consumer to add param Javadocs.
         Consumer<JavaJavadocComment> javadocCommentConsumer = comment ->
             comment.description("Creates an instance of " + model.getName() + " class.");
 
+        final int constructorPropertiesStringBuilderCapacity = 128 * (requiredProperties.size() + requiredParentProperties.size());
+
         // Use a StringBuilder with an initial capacity of 128 times the total number of required constructor properties.
         // If there are no required constructor properties this will simply be zero and result in a no-args constructor
         // being generated.
         StringBuilder constructorProperties =
-            new StringBuilder(128 * (requiredProperties.size() + requiredParentProperties.size()));
+            new StringBuilder(constructorPropertiesStringBuilderCapacity);
 
         StringBuilder superProperties = new StringBuilder(64 * requiredParentProperties.size());
 
         if (settings.isRequiredFieldsAsConstructorArgs()) {
+            final Predicate<ClientModelProperty> isWireTypeMismatch = p -> {
```

Comment:
Could we turn this into a utility method? I imagine we have other places checking for `wireType != clientType`. Might want to design this utility method with an additional boolean parameter on whether to consider `GenericType`s as equalivalent.

Reply:
sure